Sources/Jetworking/Client/Executor/UploadExecutor/Background/BackgroundUploadExecutor.swift

```
@@ -0,0 +1,116 @@
+     *  For example, the system might delay transferring large files until the device is plugged in and connected to the network via Wi-Fi.
+     *  Default value is `false`
+     */
+    init(
+        sessionConfiguration: URLSessionConfiguration,
+        uploadExecutorDelegate: UploadExecutorDelegate,
+        backgroundIdentifier: String = "com.jamitlabs.jetworking.background",
+        isDiscretionary: Bool = false
+    ) {
+        self.sessionConfiguration = sessionConfiguration
+        self.delegate = uploadExecutorDelegate
+        self.backgroundIdentifier = backgroundIdentifier
+        self.isDiscretionary = isDiscretionary
+
+        super.init()
+    }
+
+    /**
+     * # Summary
+     *  Uploading the given request with the given session configuration on a separate background session.
+     *
+     * - Parameter request:
+     *  The request to be downloaded.
+     * - Parameter fromFile:
+     *  The path to the file to be uploaded.
+     *
+     * - Returns:
+     *  The request to be able to cancel if necessary.
+     */
+    func upload(request: URLRequest, fromFile fileURL: URL) -> CancellableRequest? {
+        let uploadTask = session.uploadTask(with: request, fromFile: fileURL)
+        uploadTask.resume()
+        return uploadTask
+    }
+
+    /**
+     * # Summary
+     *  Uploading the given request with the given session configuration on a separate background session.
+     *
+     * - Parameter request:
+     *  The request to be uploaded.
+     * - Parameter from:
+     *  The data to be uploaded.
+     *
+     * - Returns:
+     *  The request to be able to cancel if necessary.
+     */
+    func upload(request: URLRequest, from bodyData: Data) -> CancellableRequest? {
+        let fileURL = FileManager.default.temporaryDirectory.appendingPathComponent(UUID().uuidString)
+        do {
+            try bodyData.write(to: fileURL)
+        } catch {
+            return nil
```

Maybe we should throw an error?

---

I would leave it as is as we do need an error handling which we might discuss in #6 
Maybe this should be the next task to start with.

---

Maybe we could leave an Todo?

---

Done.